bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,351 @@
+):
+    return {
+        'Type': 'AWS::Serverless::Function',
+        'Properties': {
+            'Runtime': py_runtime,
+            'CodeUri': deployment_name + '/',
+            'Handler': 'app.{}'.format(api_name),
+            'FunctionName': '{deployment}-{api}'.format(
+                deployment=deployment_name, api=api_name
+            ),
+            'Timeout': timeout,
+            'MemorySize': memory_size,
+            'Events': {
+                'Api': {
+                    'Type': 'Api',
+                    'Properties': {'Path': '/{}'.format(api_name), 'Method': 'post'},
+                }
+            },
+            'Policies': [{'S3ReadPolicy': {'BucketName': artifact_bucket_name}}],
+        },
+    }
+
+
+def generate_aws_lambda_template_config(
+    project_dir,
+    deployment_name,
+    api_names,
+    s3_bucket_name,
+    py_runtime,
+    memory_size,
+    timeout,
+):
+    template_file_path = os.path.join(project_dir, 'template.yaml')
+    yaml = YAML()
+    sam_config = {
+        'AWSTemplateFormatVersion': '2010-09-09',
+        'Transform': 'AWS::Serverless-2016-10-31',
+        'Globals': {
+            'Function': {'Timeout': timeout, 'Runtime': py_runtime},
+            'Api': {
+                'BinaryMediaTypes': ['image~1*'],
+                'Cors': {'AllowOrigin': "'*'"},
+                'Auth': {
+                    'ApiKeyRequired': False,
+                    'DefaultAuthorizer': 'NONE',
+                    'AddDefaultAuthorizerToCorsPreflight': False,
+                },
+            },
+        },
+        'Resources': {},
+    }
+    for api_name in api_names:
+        sam_config['Resources'][api_name] = generate_function_resource(
+            deployment_name,
+            api_name,
+            s3_bucket_name,
+            py_runtime,
+            memory_size=memory_size,
+            timeout=timeout,
+        )
+
+    yaml.dump(sam_config, Path(template_file_path))
+
+    # We add Outputs section separately, because the value should not have 'around !Sub
+    with open(os.path.join(project_dir, 'template.yaml'), 'a') as f:
+        f.write(
+            """\
+Outputs:
+    EndpointUrl:
+        Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod"
+        Description: URL for endpoint
+"""
+        )
+
+
+class AwsLambdaDeploymentOperator(DeploymentOperatorBase):
+    def apply(self, deployment_pb, yatai_service, prev_deployment):
+        try:
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+
+            bento_pb = yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+            if bento_pb.bento.uri.type != BentoUri.LOCAL:
```

Comment:
support for s3 is added recently, take a look at the previous aws lambda implementation